runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -321,83 +349,79 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 	return spine;
 }
 
-#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
-#if !((defined(LINUX) || defined(OSX)) && defined(J9VM_ENV_DATA64))
-/* Double map is only supported on LINUX 64 bit Systems for now */
-#error "Platform not supported by Double Map API"
-#endif /* !((defined(LINUX) || defined(OSX)) && defined(J9VM_ENV_DATA64)) */
-void * 
-MM_IndexableObjectAllocationModel::doubleMapArraylets(MM_EnvironmentBase *env, J9Object *objectPtr, void *preferredAddress)
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+MMINLINE J9IndexableObject *
+MM_IndexableObjectAllocationModel::getSparseAddressAndDecommitLeaves(MM_EnvironmentBase *env, J9IndexableObject *spine)
 {
-	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(env);
-	J9JavaVM *javaVM = extensions->getJavaVM();
+	Assert_MM_true(_numberOfArraylets == _allocateDescription.getNumArraylets());
 
-	GC_ArrayletLeafIterator arrayletLeafIterator(javaVM, (J9IndexableObject *)objectPtr);
-	MM_Heap *heap = extensions->getHeap();
-	UDATA arrayletLeafSize = env->getOmrVM()->_arrayletLeafSize;
+	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(env);
+	GC_ArrayObjectModel *indexableObjectModel = &extensions->indexableObjectModel;
+	const UDATA arrayletLeafSize = env->getOmrVM()->_arrayletLeafSize;
+	uintptr_t byteAmount = 0;
 	UDATA arrayletLeafCount = MM_Math::roundToCeiling(arrayletLeafSize, _dataSize) / arrayletLeafSize;
-	Trc_MM_double_map_Entry(env->getLanguageVMThread(), (void *)objectPtr, arrayletLeafSize, arrayletLeafCount);
-
-	void *result = NULL;
-
-#define ARRAYLET_ALLOC_THRESHOLD 64
-	void *leaves[ARRAYLET_ALLOC_THRESHOLD];
-	void **arrayletLeaveAddrs = leaves;
-	if (arrayletLeafCount > ARRAYLET_ALLOC_THRESHOLD) {
-		arrayletLeaveAddrs = (void **)env->getForge()->allocate(arrayletLeafCount * sizeof(uintptr_t), MM_AllocationCategory::GC_HEAP, J9_GET_CALLSITE());
-	}
 
-	if (NULL == arrayletLeaveAddrs) {
-		return NULL;
-	}
+	/* determine how many bytes to allocate outside of the spine (in arraylet leaves) */
+	Assert_MM_true(_allocateDescription.getBytesRequested() >= _allocateDescription.getContiguousBytes());
+	uintptr_t bytesRemaining = _allocateDescription.getBytesRequested() - _allocateDescription.getContiguousBytes();
 
-	GC_SlotObject *slotObject = NULL;
-	uintptr_t count = 0;
+	/* allocate leaf for each arraylet and attach it to its leaf pointer in the spine */
+	uintptr_t arrayoidIndex = 0;
+	Trc_MM_getSparseAddressAndDecommitLeaves_Entry(env->getLanguageVMThread(), spine, (void *)bytesRemaining, arrayletLeafCount, (void *)arrayletLeafSize);
+	while (0 < bytesRemaining) {
+		/* allocate the next arraylet leaf - leaves are allocated solely for the purpose of decommitting the memory later on in this function */
+		void *leaf = env->_objectAllocationInterface->allocateArrayletLeaf(env, &_allocateDescription,
+				_allocateDescription.getMemorySpace(), true);
```

Comment:
According Coding Standards the split between these two lines should be after opening bracket:
```
    void *leaf = env->_objectAllocationInterface->allocateArrayletLeaf(
        env, &_allocateDescription, _allocateDescription.getMemorySpace(), true);
```
I am not insisting to do this (and not is going to mention this in different places), but you can use common sense and apply such change if code looks better.